language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/MatchedExpressionNodeResolver.java

```diff
@@ -46,6 +49,26 @@ public MatchedExpressionNodeResolver(Node matchedNode) {
         this.matchedNode = matchedNode;
     }
 
+    /**
+     * Given the node, this method returns the optional expression in which the provided node is located.
+     *
+     * @param node Node
+     * @return Optional enclosing expression node
+     */
+    public Optional<ExpressionNode> findExpression(Node node) {
+        if (node == null) {
+            return Optional.empty();
+        }
+
+        Optional<ExpressionNode> exprNode = node.apply(this);
+        // Due to the way apply() method is implemented in some cases, this can return null
+        if (exprNode == null) {
```

Comment:
Shouldn't we fix this bug in a more higher level? For me this feels like sort of a specific remedy we are trying to provide. This same can lead to the same bug in other places as well. WDYT?

Reply:
The issue is, `apply()` method in `Node` has the following signature. 
```
public abstract <T> T apply(NodeTransformer<T> transformer);
```
As per generics, `Optional` is just another type for `T`. That's why I thought we cannot handle this better. WDYT?

Reply:
Given the implementation, that makes sense. But I think, if we call the accept method inside the at the particular node without returning null, this could have been fixed. But, yeah let's proceed with the fix for this time and let's make sure to check for the proper fix if possible as well.